lib/oli_web/components/delivery/course_content.ex

```diff
@@ -17,6 +17,30 @@ defmodule OliWeb.Components.Delivery.CourseContent do
   attr(:is_instructor, :boolean, default: false)
   attr(:preview_mode, :boolean, default: false)
 
+  def adjust_hierarchy_for_only_pages(hierarchy) do
+    case Enum.all?(hierarchy["children"], fn child -> child["type"] == "page" end) do
+      true ->
+        %{
+          "children" => [
+            %{
+              "graded" => "false",
+              "id" => "0",
+              "index" => "0",
+              "level" => "-1", # The course content browser handles case of the special level of -1
+              "next" => nil,
+              "prev" => nil,
+              "slug" => nil,
+              "title" => "Curriculum",
+              "type" => "container",
+              "children" => hierarchy["children"]
+            }
+          ]
+        }
+
+      _ -> hierarchy
+    end
```

Comment:
Shouldn't we also consider the case where the course has units/modules AND pages?